drivers/cc2420/cc2420.c

```diff
@@ -87,7 +87,17 @@ int cc2420_init(cc2420_t *dev)
     cc2420_reg_write(dev, CC2420_REG_MDMCTRL0, reg);
 
     /* go into RX state */
-    cc2420_set_state(dev, CC2420_GOTO_RX);
+    /* this doesn't compile, compiler complains that the enum type of
+     * CC2420_GOTO_RX cannot be converted to netopt_t as required by the
+     * function.
+     * The numerical value of CC2420_GOTO_RX would be NETOPT_ADDRESS (2), which
+     * can't be right.
+     * cc2420_set_state(dev, NETOPT_STATE_RX) in turn is not implemented
+     * (returns -ENOTSUP), so this was a noop for a while.
+     * TODO: revisit and confirm.
+     * (Kaspar 07/2020)
+     */
+    //cc2420_set_state(dev, CC2420_GOTO_RX);
```

Comment:
```suggestion
    cc2420_set_state(dev, NETOPT_STATE_IDLE);
```

`cc2420_set_state()` is expecting `netopt_state_t` as 2nd arg, not `netopt_t`. So when converting the numerical value of `CC2420_GOTO_RX` (2) to `netopt_state_t`, the value is `NETOPT_STATE_IDLE`, which is implemented. This behavior does indeed make sense, as this brings the device to RX.

Reply:
> `cc2420_set_state()` is expecting `netopt_state_t` as 2nd arg, not `netopt_t`.

totally right.